aten/src/ATen/native/nested/NestedTensorMath.h

```diff
@@ -21,11 +21,11 @@ inline at::Tensor wrap_buffer(at::Tensor buffer, at::Tensor nested_size_tensor)
 
 inline at::Tensor wrap_buffer(
     at::Tensor buffer, at::Tensor nested_size_tensor,
-    at::Tensor nested_stride_tensor, const std::vector<int64_t>& offsets) {
+    at::Tensor nested_stride_tensor, std::vector<int64_t>&& offsets) {
   TORCH_INTERNAL_ASSERT_DEBUG_ONLY(buffer.is_contiguous(), "Given buffer must be contiguous.");
   return at::detail::make_tensor<NestedTensorImpl>(
       std::move(buffer), std::move(nested_size_tensor),
-      std::move(nested_stride_tensor), offsets);
+      std::move(nested_stride_tensor), std::move(offsets));
```

Comment:
Tiny nit / callout: now that `offsets` is taken by rvalue reference, that prevents you from writing code like:
```
std::vector offsets = {1, 2, 3};
out = at::native::wrap_buffer(buffer, nested_size_tensor, nested_stride_tensor, offsets);
```

Maybe that's fine though? Since this function is mostly just wrapped by higher level API's that users see.

Reply:
yeah I am fairly confident that I want:
```
std::vector offsets = {1, 2, 3};
out = at::native::wrap_buffer(buffer, nested_size_tensor, nested_stride_tensor, std::move(offsets));
```
imo this makes it very explicit to the users of wrap_buffer that you are not allowed to own the offsets that gets used. If you do want to own the offsets then you need to make a copy:

```
std::vector offsets = {1, 2, 3};
out = at::native::wrap_buffer(buffer, nested_size_tensor, nested_stride_tensor, std::vector<int64_t>(offsets));
```